### PR TITLE
Handle boolean and categorical features for time series

### LINKED ISSUE #3077: Time series pipelines cannot handle boolean columns
Repro

```python
from evalml.automl import AutoMLSearch
from evalml.demos import load_weather
from evalml.automl.callbacks import raise_error_callback
import pandas as pd
import pytest

X, y = load_weather()
X.ww["bool_feature"] = pd.Series([True, False]).sample(n=X.shape[0], replace=True).reset_index(drop=True)

automl = AutoMLSearch(X, y, "time series regression",
                      problem_configuration={"max_delay": 7, "gap": 0, "forecast_horizon": 7,
                                             "date_index": "Date"},
                      error_callback=raise_error_callback,
                      max_iterations=5,
                      verbose=True)

with pytest.raises(ValueError):
    automl.search()
```

The problem is that features created from Bool/Categorical in the `DelayedFeatureTransformer` are being treated as categorical when they should be numeric.

### LINKED ISSUE #3082: One Hot Encoder errors out if there are missing values even if there are no categorical features
Repro

```python
import pandas as pd
import woodwork as ww
from evalml.pipelines.components import OneHotEncoder
import pytest


df = pd.DataFrame({"a": [1.2, 2.3, 4.5, 6.7],
                   "b": [True, False, True, True],
                   "c": [4.5, 8.3, None, 4.3]})
df.ww.init(logical_types={"a": "Double",
                          "b": "Boolean",
                          "c": "Double"})

with pytest.raises(ValueError, match="Input contains NaN"):
    OneHotEncoder().fit_transform(df)
```

I would expect this to be a no-op since there are no categorical features in the data.

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #3077, Fixes #3082

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/tests/component_tests/test_one_hot_encoder.py [hunk 1]
```diff
@@ -144,9 +144,9 @@ def test_null_values_in_dataframe():
             "col_4": [2, 0, 1, 3, 0, 1, 2],
         }
     )
-
+    X.ww.init(logical_types={"col_1": "categorical"})
```
Comment (freddyaboulton) [inline evalml/tests/component_tests/test_one_hot_encoder.py:147]: These logical types were treated as Unknown before.

File: evalml/tests/integration_tests/test_time_series_integration.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+@pytest.mark.parametrize(
+    "problem_type",
+    [
+        ProblemTypes.TIME_SERIES_BINARY,
+        ProblemTypes.TIME_SERIES_MULTICLASS,
+        ProblemTypes.TIME_SERIES_REGRESSION,
+    ],
+)
+def test_can_run_automl_for_time_series_with_categorical_and_boolean_features(
```
Comment (freddyaboulton) [inline evalml/tests/integration_tests/test_time_series_integration.py:16]: 18 second run time seems reasonable to me

File: evalml/tests/component_tests/test_one_hot_encoder.py [hunk 3]
```diff
@@ -144,9 +144,9 @@ def test_null_values_in_dataframe():
             "col_4": [2, 0, 1, 3, 0, 1, 2],
```
Comment (jeremyliweishih) [inline evalml/tests/component_tests/test_one_hot_encoder.py:144]: is there a test case where `encoder_error = OneHotEncoder(handle_missing="error")` and it fits on a data frame with a numeric column with a null value?

**Review by jeremyliweishih — APPROVED** [review]
This looks great to me! Just have a question about potentially adding a test case for the numeric OHE case you added.

**Pull request merged**